### DISCUSSION TITLE
Thank-you for all your hard work - FU HAIER!

### DISCUSSION THREAD

**Discussion by GeneralGeeCo** [Andre0512/hon#149] [General]
I can't believe the big FU Haier just sent to all it's customers, supporters and the open source community which supported them.  The short sighted nature of this really shows how out of touch they are.  
Most customers will not use the remote control feature of their appliance... of those that are tech savvy enough, 90% will use the Haier app listed in the instructions.  The 10% that will lean towards the open source version, do free product testing and be a free test/development source for Haier, pushing the boundaries on what can be done, is completely lost now.  
Look at what https://magiclantern.fm/ did for Canon cameras...  Haier - you really fucked up this time.

**Comment by jonathonbarton** [comment]
Haier-brands were on my list for replacements when we decide to upgrade from our "builder-grade" appliances in the near future - **specifically** because these two repos exist so I can connect to my home automation system.

Not any more.

I know I'm not even a rounding error in their financial health, but...now their competitors get my money.

**Reply by the-jameen** [reply]
I've recently purchased 3 Haier brand appliances due to the ease of integration these repos provide. 

The takedown notice is completely out of touch with reality - Haier doesn't profit from a subscription service or some such that is being bypassed, they are in no way losing any money because this integration exists. In my case at least, its the complete opposite, I purchased these products solely because of this integration.

Large corporations going after open source providers doesn't usually work out very well for them. I guess time will tell.

At the very least, the code will (and has been) forked so it will not be truly lost

**Reply by GeneralGeeCo** [reply]
Completely agree, the subscription service users they might be losing is a rounding error at best.  Losing the innovation and "zero-risk testing and development", along with the free advertising the open-source community was trying to provide....  what c-level idiot is making this bad decisions?

**Reply by rchilro** [reply]
Hi all, please forward on the email to haier that i will necer ever again buy a hoover/candy/haier appliance because of this. I changer my order to some air conditioning due to this atitude from haier. 
"Also will tell everyone i know not to buy. 
The best publicity is the one that comes from someone you know"
Please inform them about it. I already did.

Really sorry for this! Great work you had here!
KR

**Comment by davystrong** [comment]
Yes, thanks for all the hard work, and thanks for sharing the letter. As someone who is just starting to set up my home with Home Assistant, I know what company I'll be avoiding!

**Comment by WizardLemur** [comment]
Seriously screw Haier. It is such bs that a company is able to just shift around their monetary mass and remove end users freedom and choice with their own purchased property. I will always remember this one, that's for certain. I won't ever give a dime to these Haier a**holes.

**Comment by markterluun** [comment]
Just when I'm about to buy some of their products.. I guess I'll have to look a bit further. No way I'm going to use their own pathetic little software, the only reason they want us to is so they can collect even more of our data. F corporate greed and **FU Haier!**

**Comment by SmartLiving-Rocks** [comment]
Here to support you. Data Privacy is a Human right! Thanks for your work!
